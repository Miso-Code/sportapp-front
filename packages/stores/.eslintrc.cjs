const eslintDefaultConfig = require('@sportapp/eslint-config/library')

module.exports = {
	...eslintDefaultConfig
}
