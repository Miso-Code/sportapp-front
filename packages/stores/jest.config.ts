export default {
	preset: 'ts-jest',
	setupFilesAfterEnv: ['<rootDir>/config/test/jest.setup.ts'],
	testEnvironment: 'jsdom',
	transform: {
		'^.+\\.tsx?$': ['ts-jest']
	},
	moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
	collectCoverage: true,
	collectCoverageFrom: [
		'<rootDir>/src/**/*.{ts,tsx}',
		'!<rootDir>/src/**/interfaces/**/*.{ts,tsx}',
		'!<rootDir>/src/**/__mocks__/**/*.{ts,tsx}'
	],
	moduleNameMapper: {
		'\\.(gif|ttf|eot|svg|png|jpg)$':
			'<rootDir>/config/test/__mocks__/fileMock.ts',
		'^.+\\.(css|less|scss)$': 'babel-jest',
		'^@/(.*)$': ['<rootDir>/src/$1']
	},
	coverageThreshold: {
		global: {
			branches: 70,
			functions: 70,
			lines: 70,
			statements: 70
		}
	},
	reporters: ['default']
}
