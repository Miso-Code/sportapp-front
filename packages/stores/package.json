{
	"name": "@sportapp/stores",
	"version": "1.0.0",
	"license": "MIT",
	"main": "src/index.ts",
	"type": "module",
	"scripts": {
		"format": "prettier --write \"**/*.{ts,tsx,md,js}\"",
		"lint": "yarn lint:js",
		"lint:js": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
		"pre-commit": "lint-staged",
		"test": "jest --ci --reporters=default --config jest.config.ts --no-cache --coverage --forceExit --detectOpenHandles --logHeapUsage --silent=false --verbose"
	},
	"dependencies": {
		"@react-native-async-storage/async-storage": "^1.23.1",
		"@sportapp/sportapp-repository": "*",
		"@tanstack/react-query": "^5.28.9",
		"@testing-library/jest-dom": "^6.4.2",
		"@testing-library/react": "^14.2.2",
		"@types/jest": "^29.5.12",
		"babel-jest": "^29.7.0",
		"jest": "^29.7.0",
		"prettier": "^3.2.5",
		"react": "^18.2.0",
		"ts-jest": "^29.1.2",
		"ts-jest-mock-import-meta": "^1.2.0",
		"ts-node": "^10.9.2",
		"zustand": "^4.5.2"
	},
	"devDependencies": {
		"@sportapp/eslint-config": "*",
		"@sportapp/prettier-config": "*",
		"@sportapp/typescript-config": "*",
		"eslint": "^8.19.0",
		"lint-staged": "^15.2.2",
		"simple-zustand-devtools": "^1.1.0"
	}
}
