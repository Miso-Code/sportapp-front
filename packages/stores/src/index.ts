export { useAuthStore, initialAuthState } from './auth'
export { useSportSessionStore, initialSportSessionState } from './sportSession'
export { useSportStore, initialSportState } from './sport'
export { useUserStore, initialUserState } from './user'
export {
	usePartnerAuthStore,
	initialAuthState as initialAuthStatePartner
} from './partner/auth'
export {
	useBusinessPartnerStore,
	initialBusinessPartnerState
} from './businessPartner'
export { useAlertStore, initialAlertState } from './alert'
export {
	initialProductPartnerState,
	usePartnerProductStore
} from './partner/products'

export { initialTrainingPlanState, useTrainingPlanStore } from './trainingPlan'

export { initialPaymentPlanState, usePaymentPlanStore } from './paymentPlan'

export { initialSportsmanState, useSportsmanStore } from './sportsman'

export { initialSportEventState, useSportEventStore } from './sportEvents'
