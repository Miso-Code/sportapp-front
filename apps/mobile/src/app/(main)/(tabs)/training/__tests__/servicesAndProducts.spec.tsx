import React from 'react'
import renderer, { ReactTestRenderer, act } from 'react-test-renderer'
import { router } from 'expo-router'

import { useBusinessPartnerStore } from '@sportapp/stores'

import ServicesAndProducts from '../servicesAndProducts'

jest.mock('expo-router')

jest.mock('react-native-paper', () => {
	const textInput = jest.requireActual('react-native').TextInput
	return {
		...jest.requireActual('react-native-paper'),
		Portal: jest.fn(({ children }) => children),
		TextInput: textInput //prevents animation issues on jest
	}
})

jest.mock('@/components/KeyboardAvoidingDialog', () =>
	jest.fn(({ children }) => children)
)
jest.mock('@/components/ProductServiceCard', () => {
	const native = jest.requireActual('react-native')
	return jest.fn(({ children, ...props }) => (
		<native.View {...props} testID='productCard'>
			{children}
		</native.View>
	))
})

jest.mock('@/hooks/useDebounce', () => ({
	useDebounce: jest.fn((value) => value)
}))

jest.mock('@sportapp/stores', () => ({
	useBusinessPartnerStore: jest.fn().mockReturnValue({
		getAvailableProducts: jest.fn().mockResolvedValue([
			{
				product_id: 'product_id',
				category: 'category',
				name: 'name',
				url: 'url',
				price: 100,
				payment_type: 'payment_type',
				payment_frequency: 'payment_frequency',
				image_url: 'image_url',
				description: 'description',
				active: true
			},
			{
				product_id: 'product_id2',
				category: 'category2',
				name: 'name2',
				url: 'url2',
				price: 200,
				payment_type: 'payment_type2',
				payment_frequency: 'payment_frequency2',
				image_url: 'image_url2',
				description: 'description2',
				active: true
			}
		]),
		setProductToCheckout: jest.fn()
	})
}))

describe('ServicesAndProducts', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(<ServicesAndProducts />)
	})

	afterEach(() => {
		component.unmount()
		jest.clearAllMocks()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render an activity indicator on loading', () => {
		const [activity] = component.root.findAllByProps({
			testID: 'progressBar'
		})
		expect(activity).toBeDefined()
	})

	it('should render the products and services', async () => {
		await act(async () => {
			await Promise.resolve()
		})
		const products = component.root.findAllByProps({
			testID: 'productCard'
		})
		expect(products.length).toBe(6)
	})

	it('should render an activity indicator on scroll', async () => {
		await act(async () => {
			await Promise.resolve()
		})
		const scrollView = component.root.findByProps({
			testID: 'scrollView'
		})
		act(() => {
			scrollView.props.onScroll({
				nativeEvent: {
					contentOffset: {
						y: 100
					},
					layoutMeasurement: {
						height: 100
					},
					contentSize: {
						height: 200
					}
				}
			})
		})

		const [activity] = component.root.findAllByProps({
			testID: 'progressBar'
		})
		expect(activity).toBeDefined()
	})

	it('should get next page on scroll', async () => {
		const scrollView = component.root.findByProps({
			testID: 'scrollView'
		})
		act(() => {
			scrollView.props.onScroll({
				nativeEvent: {
					contentOffset: {
						y: 100
					},
					layoutMeasurement: {
						height: 100
					},
					contentSize: {
						height: 200
					}
				}
			})
		})
		await act(async () => {
			await Promise.resolve()
		})

		expect(
			useBusinessPartnerStore().getAvailableProducts
		).toHaveBeenCalled() //TODO toHaveBeenCalledWith + page
	})

	it('should open the product modal onPress', async () => {
		const modal = component.root.findByProps({
			testID: 'productModal'
		})

		expect(modal.props.visible).toBe(false)

		await act(async () => {
			await Promise.resolve()
		})
		const [product] = component.root.findAllByProps({
			testID: 'productCard'
		})
		act(() => {
			product.props.onPress()
		})

		expect(modal.props.visible).toBe(true)
	})

	it('should hide the product modal on dismiss', async () => {
		const modal = component.root.findByProps({
			testID: 'productModal'
		})

		await act(async () => {
			await Promise.resolve()
		})
		const [product] = component.root.findAllByProps({
			testID: 'productCard'
		})
		act(() => {
			product.props.onPress()
		})

		act(() => {
			modal.props.onDismiss()
		})

		expect(modal.props.visible).toBe(false)
	})

	it('should hide the modal on cancel', async () => {
		const modal = component.root.findByProps({
			testID: 'productModal'
		})

		await act(async () => {
			await Promise.resolve()
		})
		const [product] = component.root.findAllByProps({
			testID: 'productCard'
		})
		act(() => {
			product.props.onPress()
		})

		const cancelButton = modal.findByProps({
			testID: 'cancelButton'
		})

		act(() => {
			cancelButton.props.onPress()
		})

		expect(modal.props.visible).toBe(false)
	})

	it('should scroll up on FAB press', async () => {
		component.update(<ServicesAndProducts />)

		const fab = component.root.findByProps({
			testID: 'fabScrollUp'
		})

		const scrollView = component.root.findByProps({ testID: 'scrollView' })

		;(scrollView as any)._fiber.ref.current.scrollTo = jest.fn()
		await act(async () => {
			fab.props.onPress()
			await Promise.resolve()
		})

		expect(
			(scrollView as any)._fiber.ref.current.scrollTo
		).toHaveBeenCalled()
	})

	it('should search for products on search', async () => {
		const search = component.root.findByProps({
			testID: 'search'
		})

		act(() => {
			search.props.onChangeText('test')
		})

		const products = component.root.findAllByProps({
			testID: 'productCard'
		})

		expect(products.length).toBe(0)

		await act(async () => {
			await Promise.resolve()
		})

		expect(
			useBusinessPartnerStore().getAvailableProducts
		).toHaveBeenCalled()
	})

	it('should update product quantity on change', async () => {
		const modal = component.root.findByProps({
			testID: 'productModal'
		})

		await act(async () => {
			await Promise.resolve()
		})
		const [product] = component.root.findAllByProps({
			testID: 'productCard'
		})
		act(() => {
			product.props.onPress()
		})

		const quantity = modal.findByProps({
			testID: 'quantity'
		})

		act(() => {
			quantity.props.onChangeText('2')
		})

		expect(modal.findByProps({ testID: 'quantity' }).props.value).toBe('2')
	})

	it('should disable adquire button on invalid quantity', async () => {
		const modal = component.root.findByProps({
			testID: 'productModal'
		})

		await act(async () => {
			await Promise.resolve()
		})
		const [product] = component.root.findAllByProps({
			testID: 'productCard'
		})
		act(() => {
			product.props.onPress()
		})

		const quantity = modal.findByProps({
			testID: 'quantity'
		})

		act(() => {
			quantity.props.onChangeText(0)
		})

		const button = modal.findByProps({
			testID: 'adquireButton'
		})

		expect(button.props.disabled).toBe(true)
	})

	it('should navigate to checkout on adquire button press', async () => {
		const modal = component.root.findByProps({
			testID: 'productModal'
		})

		await act(async () => {
			await Promise.resolve()
		})
		const [product] = component.root.findAllByProps({
			testID: 'productCard'
		})
		act(() => {
			product.props.onPress()
		})

		const quantity = modal.findByProps({
			testID: 'quantity'
		})

		act(() => {
			quantity.props.onChangeText(10)
		})

		const adquireButton = modal.findByProps({
			testID: 'adquireButton'
		})

		act(() => {
			adquireButton.props.onPress()
		})

		expect(router.push).toHaveBeenCalledWith({
			pathname: 'training/servicesAndProductsCheckout',
			params: {
				quantity: 10
			}
		})
		expect(
			useBusinessPartnerStore().setProductToCheckout
		).toHaveBeenCalledWith({
			product_id: 'product_id',
			category: 'category',
			name: 'name',
			url: 'url',
			price: 100,
			payment_type: 'payment_type',
			payment_frequency: 'payment_frequency',
			image_url: 'image_url',
			description: 'description',
			active: true
		})
	})
})
