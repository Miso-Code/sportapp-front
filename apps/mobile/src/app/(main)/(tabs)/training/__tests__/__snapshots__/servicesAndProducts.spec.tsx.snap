// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesAndProducts should render correctly 1`] = `
[
  <View>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 22,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 28,
            },
            {
              "fontWeight": "bold",
            },
          ],
        ]
      }
    />
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        {
          "height": 200,
          "width": "100%",
        }
      }
    />
    <RCTScrollView
      contentInsetAdjustmentBehavior="automatic"
      style={
        {
          "maxHeight": "50%",
        }
      }
    >
      <View>
        <View
          style={{}}
        />
      </View>
    </RCTScrollView>
    <TextInput
      keyboardType="numeric"
      label="Cantidad"
      mode="outlined"
      onChangeText={[Function]}
      testID="quantity"
    />
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="cancelButton-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="cancelButton-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="cancelButton"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        {
                          "color": "rgba(179, 38, 30, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
                testID="cancelButton-text"
              >
                Cerrar
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="adquireButton-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="adquireButton-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="adquireButton"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        {
                          "color": "rgba(28, 27, 31, 0.38)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
                testID="adquireButton-text"
              >
                Adquirir
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "gap": 20,
        "justifyContent": "center",
        "marginHorizontal": 15,
        "paddingBottom": 50,
        "paddingTop": 110,
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={400}
    testID="scrollView"
  >
    <View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgb(238, 232, 244)",
            "borderRadius": 28,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": "100%",
          }
        }
        testID="search-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgb(238, 232, 244)",
              "borderRadius": 28,
              "flex": undefined,
              "flexDirection": "row",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="search-container"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 20,
                "height": 40,
                "margin": 6,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": 40,
              }
            }
            testID="search-icon-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "elevation": 0,
                  "flex": 1,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="search-icon-container"
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel="search"
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityTraits="button"
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "flexGrow": 1,
                        "justifyContent": "center",
                      },
                      {
                        "borderRadius": 20,
                      },
                    ],
                  ]
                }
                testID="search-icon"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰍉
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            accessibilityRole="search"
            keyboardAppearance="light"
            onChangeText={[Function]}
            placeholder="Search"
            placeholderTextColor="rgba(28, 27, 31, 1)"
            returnKeyType="search"
            selectionColor="rgba(103, 80, 164, 1)"
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "fontSize": 18,
                  "minWidth": 0,
                  "paddingLeft": 8,
                  "textAlign": "left",
                },
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 0,
                },
                {
                  "minHeight": 56,
                  "paddingLeft": 0,
                },
                undefined,
              ]
            }
            testID="search"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            pointerEvents="none"
            style={
              [
                {
                  "marginLeft": 16,
                  "position": "absolute",
                  "right": 0,
                },
                false,
              ]
            }
            testID="search-icon-wrapper"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 6,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "width": 40,
                }
              }
              testID="search-clear-icon-container-outer-layer"
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 20,
                    "borderWidth": 0,
                    "elevation": 0,
                    "flex": 1,
                    "overflow": "hidden",
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
                testID="search-clear-icon-container"
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel="clear"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityTraits="button"
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  centered={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={
                    {
                      "bottom": 6,
                      "left": 6,
                      "right": 6,
                      "top": 6,
                    }
                  }
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      [
                        {
                          "alignItems": "center",
                          "flexGrow": 1,
                          "justifyContent": "center",
                        },
                        {
                          "borderRadius": 20,
                        },
                      ],
                    ]
                  }
                  testID="search-clear-icon"
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "rgba(255, 255, 255, 0)",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "material-community",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    󰅖
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        category="category"
        image="image_url"
        onPress={[Function]}
        price={100}
        priceFrequency={false}
        testID="productCard"
        title="name"
      />
      <View
        category="category2"
        image="image_url2"
        onPress={[Function]}
        price={200}
        priceFrequency={false}
        testID="productCard"
        title="name2"
      />
    </View>
  </RCTScrollView>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(234, 221, 255, 1)",
        "borderRadius": 16,
        "bottom": 0,
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 8,
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
    testID="fabScrollUp-container-outer-layer"
  >
    <View
      collapsable={false}
      pointerEvents="auto"
      style={
        {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "flex": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 3,
        }
      }
      testID="fabScrollUp-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 16,
            },
          ]
        }
        testID="fabScrollUp"
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
          testID="fabScrollUp-content"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
              testID="cross-fade-icon-current"
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(33, 0, 93, 1)",
                      "fontSize": 24,
                    },
                    [
                      {
                        "lineHeight": 24,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ],
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰁝
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
